### PR TITLE
[ticket/13638] Prepend the assets path phpbb root

### PR DESCRIPTION
**Pull request by Nicofuma**
https://tracker.phpbb.com/browse/PHPBB3-13638

PHPBB3-13638

### CODE REVIEW

**Comment by VSEphpbb** [comment]
Hmm, curly brace coding guidelines?

**Comment by nickvergessen** [comment]
Yes, we are still "old school" there

**Comment by bantu** [comment]
Uhm. The code looks like voodoo. Can this be tested?

**Comment by bantu** [comment]
Hmm. Why did code sniffer not complain about the `{` before?

**Comment by bantu** [comment]
I think we have phpbb_realpath(). Is there relevant here?

**Comment by Nicofuma** [comment]
I have updated the comments

**Comment by Nicofuma** [comment]
bump

**Comment by Nicofuma** [comment]
@phpbb/development-team bump

**Comment by Nicofuma** [comment]
@phpbb/development-team bump I really think this need to be merged before we deploy master on area51

File: tests/template/asset_test.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+use phpbb\template\asset;
+
+class asset_test extends phpbb_test_case
```
Comment (nickvergessen) [inline tests/template/asset_test.php]: class name is "wrong"?

**Comment by nickvergessen** [comment]
tested and works.
Feel free to merge after the class name has been adjusted and travis passed :+1:

**Comment by Nicofuma** [comment]
@nickvergessen could you review again please? I had to change a few things after rebasing on master/HEAD

File: phpBB/phpbb/template/twig/environment.php [hunk 2]
```diff
@@ -106,16 +111,26 @@ public function get_phpbb_config()
 	}
 
 	/**
-	* Get the phpBB root path
-	*
-	* @return string
-	*/
+	 * Get the phpBB root path
+	 *
+	 * @return \phpbb\filesystem\filesystem
```
Comment (nickvergessen) [inline phpBB/phpbb/template/twig/environment.php]: nope

**Comment by nickvergessen** [comment]
other then the comment :+1:

File: phpBB/phpbb/template/twig/environment.php (hunk 2 above)
Comment (Nicofuma) [inline phpBB/phpbb/template/twig/environment.php]: grrr

**Pull request merged**